blueprints/codes/__init__.py

```diff
@@ -0,0 +1,56 @@
+"""Module for the abstract base class Equation."""
+from abc import ABC, abstractmethod
+
+
+class Equation(ABC):
+    """
+    Abstract base class for equations used in the code.
+    """
+
+    @property
+    def equation_number(self) -> str:
```

Comment:
I would suggest: equation_label or something, it is not always an actual 'number' in my opinion.

Reply:
I have no particular preference on this one but last meeting most people went for equation_number